packages/scm/src/browser/decorations/scm-tab-bar-decorator.ts

```diff
@@ -0,0 +1,74 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { SCM_VIEW_CONTAINER_ID } from '../scm-contribution';
+import { ScmService } from '../scm-service';
+import { TabBarDecorator } from '@theia/core/lib/browser/shell/tab-bar-decorator';
+import { Title, Widget } from '@theia/core/lib/browser';
+import { WidgetDecoration } from '@theia/core/lib/browser/widget-decoration';
+
+@injectable()
+export class ScmTabBarDecorator implements TabBarDecorator {
+
+    readonly id = 'theia-scm-tabbar-decorator';
+    protected readonly emitter = new Emitter<void>();
+
+    @inject(ScmService)
+    protected readonly scmService: ScmService;
+
+    @postConstruct()
+    protected init(): void {
+        this.scmService.onDidChangeSelectedRepository(repository => {
+            if (repository) {
+                repository.provider.onDidChange(() => this.fireDidChangeDecorations());
+            }
+        });
+    }
+
+    decorate(title: Title<Widget>): WidgetDecoration.Data[] {
+        if (title.owner.id === SCM_VIEW_CONTAINER_ID) {
+            const changes: number = this.updateBadgeCount();
+            return changes > 0 ? [{ badge: changes.toString() }] : [];
+        } else {
+            return [];
+        }
+    }
+
+    updateBadgeCount(): number {
```

Comment:
It looks very cool 👍 

I think this should not be public but protected, and checking the VS Code behavior: a badge decorator is not necessarily a number, it can be an exclamation mark: `!`, or _ work in progress_.

![Screen Shot 2020-07-10 at 09 25 14](https://user-images.githubusercontent.com/1405703/87128136-66db9580-c28f-11ea-8230-cafa462ac249.jpg)

Reply:
Let's consider renaming `updateBadgeCount` to something else; it does not update any state but returns with a number.

Reply:
it seems to be specifics of the scm implementation? badge is of `string` type: https://github.com/eclipse-theia/theia/pull/8156/files#diff-aeb0c43202ac4f2c756bd99bcbc11c73R336

Reply:
VS Code supports different [`Badge`](https://github.com/microsoft/vscode/blob/master/src/vs/workbench/services/activity/common/activity.ts#L76-L114) types, but this pull-request only plans to support "`NumberBadges`" (counters):
I think we can improve support for other types of badges in subsequent pull-requests?

Reply:
> Let's consider renaming updateBadgeCount to something else;

Can you please take care of it? `updateBadgeCount` does not update anything at all, just returns a number. Thanks!

Reply:
@kittaakos Sorry, I missed your comments. I updated the function name to `collectChangesCount`.